framework/src/base/FEProblem.C

```diff
@@ -2587,13 +2587,16 @@ FEProblem::computeUserObjects(ExecFlagType type/* = EXEC_TIMESTEP_END*/, UserObj
 }
 
 void
-FEProblem::reinitBecauseOfGhosting()
+FEProblem::reinitBecauseOfGhostingOrNewGeomObjects()
 {
-  // Need to see if _any_ processor has ghosted elems
-  dof_id_type ghosted = _ghosted_elems.size();
-  _communicator.sum(ghosted);
+  // Need to see if _any_ processor has ghosted elems or geometry objects.
+  bool needs_reinit = ! _ghosted_elems.empty();
+  needs_reinit = needs_reinit || ! _geometric_search_data._nearest_node_locators.empty();
+  needs_reinit = needs_reinit || ( _displaced_problem && ! _displaced_problem->geomSearchData()._nearest_node_locators.empty() );
+  dof_id_type do_reinit = static_cast<dof_id_type>(needs_reinit);
```

Comment:
@shanestafford, not that it's a big deal but why cast this Boolean to an integer type only to use it as a Boolean again?